### PR TITLE
[🚀 ESL Popup] add position-origin attribute

### PR DESCRIPTION
**Pull request by dshovchko**
Added a new attribute `position-origin="outer|inner"` that allows or disallows to overlap of the trigger by a popup window.

sshot_2024-11-06-12-47-20

Closes: #2746

### CODE REVIEW

File: src/modules/esl-popup/core/esl-popup-position.ts [hunk 1]
```diff
@@ -141,19 +143,51 @@ function fitOnMajorAxis(cfg: PopupPositionConfig, value: PopupPositionValue): Po
...
+  switch (position) {
+    case 'right':
+      x = (hasInnerOrigin ? inner.right : inner.x) - popup.width;
+      break;
+    case 'bottom':
+      y = (hasInnerOrigin ? inner.bottom : inner.y) - popup.height;
+      break;
+  }
+  return new Rect(x, y, popup.width, popup.height);
 }
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup-position.ts:191]: Methods are absolutely the same except invertion of `hasInnerOrigin` input value

**Review by yadamskaya — APPROVED** [review]

**Review by ala-n — APPROVED** [review]
We need to think of optimization of these utils...

File: src/modules/esl-popup/core/esl-popup.ts [hunk 2]
```diff
@@ -401,10 +407,11 @@ export class ESLPopup extends ESLToggleable {
     const triggerRect = Rect.from(this.activator).shift(window.scrollX, window.scrollY);
     const {containerRect} = this;
 
-    const innerMargin = this._offsetTrigger + arrowRect.width / 2;
+    const innerMargin = this._offsetTrigger + (this.positionOrigin === 'inner' ? 0 : arrowRect.width / 2);
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply src/modules/esl-popup/core/esl-popup.ts:410]: This is not a limitation. This is just a position adjustment.

**Comment by dshovchko** [comment]
> We need to think of optimization of these utils...

I'm going to create a few issues to this

**Review by fshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-popup/README.md [hunk 3]
```diff
@@ -89,13 +89,17 @@ You can also add additional classes and styles when activating and displaying th
 
 - `position` - popup position relative to the trigger (currently supported: 'top', 'bottom', 'left', 'right' ) ('top' by default)
   
+- `position-origin` <i class="badge badge-sup badge-warning">beta</i> - This attribute specifies from which side of the trigger grows popup to achieve the desired position ('outer' by default). Available options:
```
Comment (abarmina) [inline src/modules/esl-popup/README.md:92]: extra cosmetic
```suggestion
- `position-origin` <i class="badge badge-sup badge-warning">beta</i> - this attribute specifies from which side of the trigger grows popup to achieve the desired position ('outer' by default). Available options:
```

**Review by abarmina — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

**Pull request merged**